src/Compute/Compute/Extension/AzureDiskEncryption/Scripts/Find_1passAdeVersion_VM.ps1

```diff
@@ -1,96 +1,96 @@
-{
-    $getAllVMInSubscription = Get-AzVM
-    $outputContent = @()
-
-    foreach ($vmobject in $getAllVMInSubscription)
-    {
-        $vm_OS = ""
-        if ($vmobject.OSProfile.WindowsConfiguration -eq $null) 
-        { 
-            $vm_OS = "Linux" 
-        }
-        else 
-        {
-            $vm_OS = "Windows" 
-        }
-    
-        $vmInstanceView = Get-AzVM -ResourceGroupName $vmobject.ResourceGroupName -Name $vmobject.Name -Status    
-
-        $isVMADEEncrypted = $false
-        $isStoppedVM = $false
-        $adeVersion = ""
-
-        #Find ADE extension version if ADE extension is installed                 
-        $vmExtensions = $vmInstanceView.Extensions
-        foreach ($extension in $vmExtensions)
-        {
-            if ($extension.Name -like "azurediskencryption*")
-            {
-                $adeVersion = $extension.TypeHandlerVersion
-                $isVMADEEncrypted = $true
-                break;            
-            }            
-        }
-
-        #Look for encryption settings on disks. This applies to VMs that are in deallocated state
-        #Extension version information is unavailable for stopped VMs
-        if ($isVMADEEncrypted -eq $false)
-        {
-            $disks = $vmInstanceView.Disks
-            foreach ($diskObject in $disks)
-            {
-                if ($diskObject.EncryptionSettings -ne $null)
-                {
-                    $isStoppedEncryptedVM = $true
-                    break;
-                }
-            }
-        }        
-
-        if ($isVMADEEncrypted)
-        {        
-            #Prepare output content for single pass VMs            
-            if ((($vm_OS -eq "Windows") -and ($adeVersion -like "2.*")) -or (($vm_OS -eq "Linux") -and ($adeVersion -like "1.*")))
-            {            
-                $results = @{
-                VMName = $vmobject.Name
-                ResourceGroupName = $vmobject.ResourceGroupName
-                VM_OS = $vm_OS
-                ADE_Version = $adeVersion                        
-                }
-                $outputContent += New-Object PSObject -Property $results
-                Write-Host "Added details for encrypted VM " $vmobject.Name
-            }               
-        }
-        elseif ($isStoppedEncryptedVM)
-        {
-            $results = @{
-                VMName = $vmobject.Name
-                ResourceGroupName = $vmobject.ResourceGroupName
-                VM_OS = $vm_OS
-                ADE_Version = "Not Available"                        
-                }
-                $outputContent += New-Object PSObject -Property $results
-                Write-Host "Added details for encrypted VM. ADE version = Not available " $vmobject.Name
-        }                      
-   }
-
-    #Write to output file
-    $filePath = ".\" + $SubscriptionId + "_AdeVMInfo.csv"
-    $outputContent | export-csv -Path $filePath -NoTypeInformation
+﻿<#
```

Comment:
Is it showing up as diff because of indentation?

Reply:
I'm not sure. I tried really hard to remove this from the PR as well but git wouldn't let me. Taking all suggestions here...

Reply:
VS may be adding a BOM. It may be harmless and not required to make any changes. Here's an instruction about avoiding adding BOM to the file in VS for future: https://vlasovstudio.com/fix-file-encoding/